### PR TITLE
API Changes

### PR DESCRIPTION
**Pull request by devkiran**

### CODE REVIEW

**Comment by versecafe** [comment]
@devkiran Would moving to drizzle make sense if already adding zod wrappers? fix prismas performance issues and [dirzzle-zod](https://orm.drizzle.team/docs/zod) makes fusing the two to generate type validation from db models easier

File: apps/docs/api-reference/errors.mdx
Comment (steven-tey) [inline apps/docs/api-reference/errors.mdx:1]: Can we add the status codes for each of these responses here? E.g. 403, 404, 400, etc.

File: apps/web/app/api/analytics/[endpoint]/route.ts
Comment (steven-tey) [inline apps/web/app/api/analytics/[endpoint]/route.ts:1]: this looks great – can we get this added to `/api/analytics/edge/[endpoint]/route.ts` as well?

File: apps/web/app/api/cron/links/event/route.ts
Comment (steven-tey) [inline apps/web/app/api/cron/links/event/route.ts:1]: Why do we need to include `excludeRoot` here if it's an optional param?

File: apps/web/app/api/links/route.ts [hunk 1]
```diff
@@ -1,89 +1,106 @@
...
+          throw new DubApiError({
+            code: "rate_limit_exceeded",
+            message:
+              "Rate limited – you can only create up to 10 links per day without an account.",
+          });
+        }
       }
-    }
 
-    const { link, error, status } = await processLink({
```
Comment (steven-tey) [inline apps/web/app/api/links/route.ts:59]: what happened to `status`? should we remove it and fully rely on `code`?

File: apps/web/app/api/og/analytics/route.tsx [hunk 2]
```diff
@@ -36,6 +36,7 @@ export async function GET(req: NextRequest) {
     linkId: link.id,
     endpoint: "timeseries",
     interval: "30d",
+    excludeRoot: false
```
Comment (steven-tey) [inline apps/web/app/api/og/analytics/route.tsx]: same question about `excludeRoot` – why do we need to define it? can it just be left undefined

**Review by steven-tey — CHANGES_REQUESTED** [review]

File: apps/docs/api-reference/errors.mdx
Comment (devkiran) [reply apps/docs/api-reference/errors.mdx:1]: Yup, I'll add status codes.

File: apps/web/app/api/analytics/[endpoint]/route.ts
Comment (devkiran) [reply apps/web/app/api/analytics/[endpoint]/route.ts:1]: Sure.

File: apps/web/app/api/cron/links/event/route.ts
Comment (devkiran) [reply apps/web/app/api/cron/links/event/route.ts:1]: Good catch, can fix it with some tweak in zod schema.

File: apps/web/app/api/og/analytics/route.tsx (hunk 2 above)
Comment (devkiran) [reply apps/web/app/api/og/analytics/route.tsx]: Yup, it can left undefined

File: apps/web/app/api/links/route.ts (hunk 1 above)
Comment (devkiran) [reply apps/web/app/api/links/route.ts:59]: `status` was replaced with error `code`, no longer needed. I think we can safely remove `status`.

**Review by steven-tey — APPROVED** [review]

**Pull request merged**